### PR TITLE
Add new item that reduces warp by one

### PR DESCRIPTION
**Pull request by Prometheus0000**
It's chance based, see https://github.com/GTNewHorizons/NewHorizons/issues/5643. Numbers are based on the wither getting moved back (whenever that happens), I'd like to make it configurable, but I don't know how. Not sure I got the research to show up correctly ingame, but works when I use a script file (grab it from the issue page). Uses a copy of the pure tear recipe, since we're just changing it anyway.

### CODE REVIEW

File: src/main/java/shukaro/warptheory/items/itemCleanserMinor.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import shukaro.warptheory.WarpTheory;
+import shukaro.warptheory.handlers.WarpHandler;
+import shukaro.warptheory.util.ChatHelper;
+import shukaro.warptheory.util.Constants;
+import shukaro.warptheory.util.FormatCodes;
+
+import java.util.List;
+import java.util.Locale;
+
+public class ItemCleanserMinor extends Item
```
Comment (mitchej123) [inline src/main/java/shukaro/warptheory/items/itemCleanserMinor.java]: Why not just extend from ItemCleanser and override the methods that you change? I"m assuming most of them are going to be pretty similar.

File: src/main/java/shukaro/warptheory/handlers/WarpHandler.java [hunk 2]
```diff
@@ -181,6 +184,22 @@ public static void purgeWarp(EntityPlayer player)
         removeWarp(player, getTotalWarp(player));
     }
 
+	//add function to remove 1 warp, chance based
+	//warp < 50 = 0%, warp >= 150 = 100%
+    public static void purgeWarpMinor(EntityPlayer player)
+    {
+		Random ran = new Random();
```
Comment (draknyte1) [inline src/main/java/shukaro/warptheory/handlers/WarpHandler.java]: No point allocating a new Random instance every single time..
Add it as a class member `private static final chanceRandom = new Random();`

**Review by draknyte1 — CHANGES_REQUESTED** [review]
While the thought is good, I’ve got some suggestions:
1) the indentation is absolutely horrible. 
2) learn how basic java works, extend the original item and then override the relevant methods, as suggested by @mitchej123.
3) don’t think the super() call in the items constructor is necessary. 
4) pure? tear seems silly.. call it an impure tear imho.

**Comment by Prometheus0000** [comment]
1) Did I not do them right? I thought I did. Maybe it got messed up.
2) I'm not a coder, I've never taken a Java class. If I do this again, I'll try the extending thing.
3) ? I didn't type/use super anywhere? I saw it on the page mitch linked, but I couldn't really get it. When it started talking about sub-classes and superclasses, my eyes kinda glazed over.
4) I was thinking of it as fake of the real one.

**Comment by draknyte1** [comment]
It’s not a fake though, it’s just an inferior version.. 
if it was fake it wouldn’t do anything at all.

**Comment by Prometheus0000** [comment]
I don't really care, if other people want it changed, that's fine by me. Maybe bootleg ripoff version would be a better descriptor?

**Comment by draknyte1** [comment]
Impure still makes more sense for that description...

**Comment by yukieiji** [comment]
The addition of the new item in the original issue is opposed by richardhendricks and I think there isn't enough discussion about the new item (specifications, recipes, whether to add, etc. ).

**Comment by mitchej123** [comment]
> The addition of the new item in the original issue is opposed by richardhendricks and I think there isn't enough discussion about the new item (specifications, recipes, whether to add, etc. ).

I disagree with this sentiment.  My reading of his comment is that he was opposed to it being guaranteed 1-100; but will let him chime in.

Also, what a perfect place to chime in on the things you're talking about... on one of the PR's starting to implement it!  (I'd advocate for action vs inaction)

**Comment by yukieiji** [comment]
I didn't understand the purpose and function of the item well.
Now that I understand, I'm in favor of adding this item.

**Review by repo-alt — APPROVED** [review]

**Pull request merged**